homeassistant/components/modbus/sensor.py

```diff
@@ -131,18 +131,22 @@ class SlaveSensor(
     RestoreEntity,
     SensorEntity,
 ):
-    """Modbus slave binary sensor."""
+    """Modbus slave register sensor."""
 
     def __init__(
         self,
         coordinator: DataUpdateCoordinator[list[int] | None],
         idx: int,
         entry: dict[str, Any],
     ) -> None:
-        """Initialize the Modbus binary sensor."""
+        """Initialize the Modbus register sensor."""
         idx += 1
         self._idx = idx
         self._attr_name = f"{entry[CONF_NAME]} {idx}"
+        self._attr_unique_id = entry.get(CONF_UNIQUE_ID)
```

Comment:
I think this should be done for binary sensor as well (it allows you to name your slave sensors and assign them to different areas etc),  but I am trying to keep this file targeted as I am not set up to test binary sensor. I will leave that for another dev.

Reply:
A UNIQUE_ID is as the names indicates unique and non-changeable, because of that it is not allowed to make it configurable.

Reply:
Its already configurable for the non-slave ... this just inherits it to the slave (which was not correctly happening before):
https://www.home-assistant.io/integrations/modbus/
unique_id string (optional)
An ID that uniquely identifies this sensor. If two sensors have the same unique ID, Home Assistant will raise an exception.

Reply:
responded to earlier.